src/ode_solver/runge_kutta_ode_solver.cpp

```diff
@@ -114,6 +118,12 @@ void RungeKuttaODESolver<dim,real,n_rk_stages,MeshType>::allocate_ode_system ()
     if(this->all_parameters->use_inverse_mass_on_the_fly == false) {
         this->pcout << " evaluating inverse mass matrix..." << std::flush;
         this->dg->evaluate_mass_matrices(true); // creates and stores global inverse mass matrix
+        //RRK needs both mass matrix and inverse mass matrix
+        using ODEEnum = Parameters::ODESolverParam::ODESolverEnum;
+        ODEEnum ode_type = this->ode_param.ode_solver_type;
+        if (ode_type == ODEEnum::rrk_explicit_solver){
+            this->dg->evaluate_mass_matrices(false); // creates and stores global mass matrix
```

Comment:
why is `evaluate_mass_matrices` called twice?

Reply:
Because using true or false toggles whether the mass matrix or inverse mass matrix is called. As indicated in the comments, passing `true` results in the inverse mass matrix being stored, and passing `false` results in the mass matrix being stored. They are stored in different variables so there's no issue with calling it twice.

Reply:
Ahh okay got it, thanks!